src/backend/kms/surface/mod.rs

```diff
@@ -364,17 +369,24 @@ impl Surface {
 
     pub fn add_node(&mut self, node: DrmNode, gbm: GbmAllocator<DrmDeviceFd>, egl: EGLContext) {
         self.known_nodes.insert(node);
-        let _ = self
-            .thread_command
-            .send(ThreadCommand::NodeAdded { node, gbm, egl });
+        let (tx, rx) = std::sync::mpsc::sync_channel(1);
+        let _ = self.thread_command.send(ThreadCommand::NodeAdded {
+            node,
+            gbm,
+            egl,
+            sync: tx,
+        });
+        let _ = rx.recv();
```

Comment:
I guess blocking here isn't really an issue since it should be uncommon to remove a device node (though not that uncommon with DisplayLink) and shouldn't block for very long. But it should probably at least have a comment explaining why we're waiting here?